apps/api/src/app/auth/auth.service.ts

```diff
@@ -40,6 +41,33 @@ export class AuthService {
     });
   }
 
+  public async validateInternetIdentityLogin(principalId: string) {
+    try {
+      const provider: Provider = 'INTERNET_IDENTITY';
+
+      let [user] = await this.userService.users({
```

Comment:
@samlinux With this implementation, it would be possible to randomly find a user by its principal id. Is it safe anyway (because e.g. the principle id is long enough)?

Reply:
Worth to note that the principal ID is generated anonymously and is different for each domain - i.e. you cannot track principal ID across websites

e.g. a user that uses Internet Identity with an anchor 1234 would get a principal ID 5678 for Ghostfolio but abcd for Papyrs

Reply:
> Added few comments, hope it helps
> 
> Cool to see II used in a web2 project as well 👍

Thank you very much for taking the time to give your feedback, @peterpeterparker! 😎